neverbleed.c

```diff
@@ -965,7 +1020,10 @@ static EVP_PKEY *ecdsa_create_pkey(neverbleed_t *nb, size_t key_index, int curve
     exdata->key_index = key_index;
 
     ec_key = EC_KEY_new_method(nb->engine);
-    EC_KEY_set_ex_data(ec_key, 0, exdata);
+    NEVERBLEED_MULTITHREAD_ONCE({
+        rsa_exdata_idx = EC_KEY_get_ex_new_index(0, NULL, NULL, NULL, rsa_exdata_free_callback);
+    });
```

Comment:
It took me a while to find out the problem above, as I could not tell immediately if the error that I saw was due to use of uninitialized value or something else.

Considering that, I think it mike make sense to create a function for getting the index, rather than exposing a variable holding the index and trying to initialize the index at the right moment. The precedence I have in mind is [get_ptls_index of h2o](https://github.com/h2o/h2o/blob/6cc4918/lib/common/socket.c#L2139-L2144).

WDYT?